src/test/java/com/commercetools/sync/types/utils/typeactionutils/BuildLocalizedEnumUpdateActionsTest.java

```diff
@@ -0,0 +1,205 @@
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            Collections.emptyList(),
+            ENUM_VALUES_ABC
+        );
+
+        assertThat(updateActions).containsExactly(
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_A),
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_B),
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_C)
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithIdenticalPlainEnum_ShouldNotBuildUpdateActions() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_ABC
+        );
+
+        assertThat(updateActions).isEmpty();
+    }
+
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithOnePlainEnumValue_ShouldBuildAddEnumValueAction() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_ABCD
+        );
+
+        assertThat(updateActions).containsExactly(
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_D)
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithOneEnumValueSwitch_ShouldBuildRemoveAndAddEnumValueActions() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_ABD
+        );
+
+        // remove enum value actions not exists for type resources
+        assertThat(updateActions).containsExactly(
+            AddLocalizedEnumValue.of(FIELD_NAME_1, ENUM_VALUE_D)
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithDifferent_ShouldBuildChangeEnumValueOrderAction() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_CAB
+        );
+
+        assertThat(updateActions).containsExactly(
+            ChangeLocalizedEnumValueOrder.of(FIELD_NAME_1, asList(
+                ENUM_VALUE_C.getKey(),
+                ENUM_VALUE_A.getKey(),
+                ENUM_VALUE_B.getKey()
+            ))
+        );
+    }
+
+    @Test
+    public void buildLocalizedEnumUpdateActions_WithRemovedAndDifferentOrder_ShouldBuildChangeOrderAndRemoveActions() {
+        final List<UpdateAction<Type>> updateActions = buildLocalizedEnumValuesUpdateActions(
+            FIELD_NAME_1,
+            ENUM_VALUES_ABC,
+            ENUM_VALUES_CB
+        );
+
+        // remove enum value actions not exists for type resources
+        assertThat(updateActions).containsExactly(
+            ChangeLocalizedEnumValueOrder.of(FIELD_NAME_1, asList(
+                ENUM_VALUE_C.getKey(),
```

Comment:
What about A? Don't you have to have all enums in this action?

Reply:
A should be removed but it's not supported by the API.

Reply:
I would keep a test that its failing for it but mark it with @Ignore annotation and a TODO comment to fix it once the API provides it.